spec/reek/smell_detectors/utility_function_spec.rb

```diff
@@ -177,6 +177,27 @@ def bravo(charlie)
     end
   end
 
+  context 'when examining class_methods block' do
+    it 'reports on the refined class' do
+      src = <<-RUBY
+        module Alpha
+          extend ActiveSupport::Concern
+
+          class_methods do
+            def bravo(arg1, arg2)
+              return 1 if arg1.something?
+              return 2 if arg2.something_else?
+
+              3
+            end
+          end
+        end
+      RUBY
+
+      expect(src).not_to reek_of(:UtilityFunction, context: 'Alpha#bravo')
+    end
+  end
```

Comment:
Can you also add a test case that shows that methods defined outside the `class_methods` block still trigger a UtilityFunction warning?

Reply:
Yup